src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -715,7 +715,7 @@ private string GeneratePropMetadataInitFunc(TypeGenerationSpec typeGenerationSpe
 
                     TypeGenerationSpec memberTypeMetadata = memberMetadata.TypeGenerationSpec;
 
-                    string clrPropertyName = memberMetadata.ClrName;
+                    string clrPropertyName = memberMetadata.ExactNameSpecifiedInSourceCode;
```

Comment:
Rename this variable and use it in place of `memberMetadata.ExactNameSpecifiedInSourceCode` references below.